### PR TITLE
Added XML documentation to the DatabaseSchemaHelper class

### PR DESCRIPTION
**Pull request by abjerner**
Not sure how many are using the reference documentation (or just browsing the source code), but noticed earlier today that the class lacked some XML documentation.

image

### CODE REVIEW

File: src/Umbraco.Core/Persistence/DatabaseSchemaHelper.cs [hunk 1]
```diff
@@ -90,20 +138,54 @@ internal void CreateDatabaseSchemaDo(IMigrationEntryService migrationEntryServic
             _logger.Info<Database>("Finalized database schema creation");
         }
 
+        /// Creates a new table in the database based on the type of <typeparamref name="T"/>.
+        ///
```
Comment (poornimanayar) [inline src/Umbraco.Core/Persistence/DatabaseSchemaHelper.cs]: Good idea to get rid of this blank line?

File: src/Umbraco.Core/Persistence/DatabaseSchemaHelper.cs [hunk 2]
```diff
@@ -90,20 +138,54 @@ internal void CreateDatabaseSchemaDo(IMigrationEntryService migrationEntryServic
...
         public void CreateTable<T>(bool overwrite)
            where T : new()
         {
             var tableType = typeof(T);
             CreateTable(overwrite, tableType);
         }
 
+        /// <summary>
+        /// Creates a new table in the database based on the type of <typeparamref name="T"/>.
+        ///
```
Comment (poornimanayar) [inline src/Umbraco.Core/Persistence/DatabaseSchemaHelper.cs]: Remove blank line [142](https://github.com/umbraco/Umbraco-CMS/pull/3205/files#diff-140ac76b9076c9248a612ce2aacb042dR142) & [146](https://github.com/umbraco/Umbraco-CMS/pull/3205/files#diff-140ac76b9076c9248a612ce2aacb042dR146) as well may be? Has this been introduced purposefully?

Comment (abjerner) [reply src/Umbraco.Core/Persistence/DatabaseSchemaHelper.cs]: Yes, I added the extra lines by purpose. Whether it makes sense may depend on the tool used to generate the reference documentation. @nul800sebastiaan do you have some hints here?

As we don't want long lines in the code, it's normal to add a single line break, which then is just interpreted as regular white space. Some tools will treat a double line break similar to Markdown, making it a paragraph (which was my intention).

None of this do however affect how it's shown in Visual Studio, as it's just shown as a single line. It appears that Visual Studio won't even honor `<br />` elements in the description.

**Comment by nul800sebastiaan** [comment]
Thanks @abjerner - yeah we use DocFX which accepts some form of MarkDown as far as I can remember. Besides I think the empty lines make it more readable anyway if you're looking at the source, so all good! 👍

File: src/Umbraco.Core/Persistence/DatabaseSchemaHelper.cs (hunk 1 above)
Comment (Shazwazza) [reply src/Umbraco.Core/Persistence/DatabaseSchemaHelper.cs]: There's also no <summary> tag here

**Pull request merged**